lib/connection.js

```diff
@@ -896,23 +896,30 @@ Connection.prototype.sobject = function(type) {
 /**
  * Get identity information of current user
  *
+ * @param {Object} [options] - Identity call options
+ * @param {Object} [options.headers] - Additional HTTP request headers sent in identity request
  * @param {Callback.<IdentityInfo>} [callback] - Callback function
  * @returns {Promise.<IdentityInfo>}
  */
-Connection.prototype.identity = function(callback) {
+Connection.prototype.identity = function(options, callback) {
+  if (typeof options === 'function') {
+    callback = options;
+    options = {};
+  }
+  options = options || {};
   var self = this;
   var idUrl = this.userInfo && this.userInfo.url;
   return Promise.resolve(
     idUrl ?
     { identity: idUrl } :
-    this.request(this._baseUrl())
+    this.request({ method: 'GET', url: this._baseUrl(), headers: options.headers })
   ).then(function(res) {
     var url = res.identity;
     url += '?format=json&oauth_token=' + encodeURIComponent(self.accessToken);
     var transport = Transport.JsonpTransport.supported ?
       new Transport.JsonpTransport('callback') :
       undefined;
-    return self.requestGet(url, { transport: transport });
+    return self.requestGet(url, options, { transport: transport });
```

Comment:
The interface of `requestGet` is `requestGet(url, options, callback)`, options to be included in 2nd arg.